### PR TITLE
code optimizations

### PR DESCRIPTION
**Pull request by deepanshutr**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and please add links to any
relevant follow-up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json)
  in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (
  see [PR Targeting](https://github.com/AssetMantle/schema/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines
  for [building modules](https://github.com/AssetMantle/schema/blob/master/docs/building-modules)
- [ ] included the necessary unit and
  integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add your handle next to the items
reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the
  correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by deepanshutr — COMMENTED** [review]
self reviewed

File: go/documents/base/document.go [hunk 1]
```diff
@@ -18,12 +19,11 @@ func (document *Document) ValidateBasic() error {
 	if err := document.ClassificationID.ValidateBasic(); err != nil {
 		return err
 	}
-	if err := document.Immutables.ValidateBasic(); err != nil {
-		return err
-	}
-	if err := document.Mutables.ValidateBasic(); err != nil {
+
+	if err := document.GetImmutables().GetImmutablePropertyList().Add(baseLists.AnyPropertiesToProperties(document.GetMutables().GetMutablePropertyList().Get()...)...).ValidateBasic(); err != nil {
```
Comment (avkr003) [inline go/documents/base/document.go:23]: Since Add updates when property is found again, it won't throw the error of duplicate

**Review by avkr003 — CHANGES_REQUESTED** [review]

File: go/data/base/string_data.go [hunk 2]
```diff
@@ -36,7 +35,13 @@ func (stringData *StringData) GetBondWeight() sdkTypes.Int {
 	return dataConstants.StringDataWeight
 }
 func (stringData *StringData) Compare(listableData data.ListableData) int {
-	return bytes.Compare(stringData.Bytes(), listableData.Bytes())
+	compareStringData := listableData.ToAnyListableData().Get().(*StringData)
```
[the remark this answers is not in the document]
Comment (deepanshutr) [reply go/data/base/string_data.go:38]: this is equivalent, checked

[the remark this answers is not in the document]
Comment (deepanshutr) [reply go/data/base/string_data.go:38]: compare gives difference, less efficient for this use case

**Review by avkr003 — APPROVED** [review]

**Pull request merged**